contracts/javascore/ibc/src/main/java/ibc/ics23/commitment/Proof.java

```diff
@@ -62,6 +65,210 @@ public static void checkAgainstSpec(ExistenceProof proof, ProofSpec spec) {
+            if (Arrays.equals(leftInnerOp.getPrefix(), rightInnerOp.getPrefix()) &&
+                    Arrays.equals(leftInnerOp.getSuffix(), rightInnerOp.getSuffix())) {
+                leftIdx = leftIdx - 1;
+                rightIdx = rightIdx - 1;
+                continue;
+            }
+            break;
+        }
+
+        if (!isLeftStep(spec, left.get(leftIdx), right.get(rightIdx))) {
+            return false;
+        }
+
+        if (!isRightMost(spec, sliceInnerOps(left, 0, leftIdx))) {
+            return false;
+        }
+
+        if (!isLeftMost(spec, sliceInnerOps(right, 0, rightIdx))) {
+            return false;
+        }
+        return true;
+    }
+
+    private static int orderFromPadding(InnerSpec spec, InnerOp op) {
+        int maxBranch = spec.getChildOrder().size();
+        for (int branch = 0; branch < maxBranch; branch++) {
+            BigInteger[] padding = getPadding(spec, branch);
+            BigInteger minPrefix = padding[0];
+            BigInteger maxPrefix = padding[1];
+            BigInteger suffix = padding[2];
+
+            if (hasPadding(op, minPrefix, maxPrefix, suffix)) {
+                return branch;
+            }
+        }
+        throw new UserRevertedException("Cannot find any valid spacing for this node");
+    }
+
+    private static BigInteger[] getPadding(InnerSpec spec, int branch) {
+        BigInteger childSize = spec.getChildSize();
+        int idx = getPosition(spec.getChildOrder(), branch);
+        BigInteger prefix = BigInteger.valueOf(idx).multiply(childSize);
+        BigInteger minPrefix = prefix.add(spec.getMinPrefixLength());
+        BigInteger maxPrefix = prefix.add(spec.getMaxPrefixLength());
+        BigInteger suffix = BigInteger.valueOf(spec.getChildOrder().size() - 1 - idx).multiply(childSize);
+
+        return new BigInteger[]{minPrefix, maxPrefix, suffix};
+    }
+
+    private static int getPosition(List<BigInteger> order, int branch) {
+        int orderLength = order.size();
+        if (branch >= orderLength) {
+            throw new UserRevertedException("Invalid branch");
+        }
+
+        for (int i = 0; i < orderLength; i++) {
+            if (order.get(i).equals(BigInteger.valueOf(branch))) {
+                return i;
+            }
+        }
+        return 0;
+    }
+
+    private static boolean hasPadding(InnerOp op, BigInteger minPrefix, BigInteger maxPrefix, BigInteger suffix) {
+        int opPrefixLength = op.getPrefix().length;
+        if (opPrefixLength < minPrefix.intValue()) return false;
+        if (opPrefixLength > maxPrefix.intValue()) return false;
+        return op.getSuffix().length == suffix.intValue();
+    }
+
+    private static List<InnerOp> sliceInnerOps(List<InnerOp> array, int start, int end) {
+        List<InnerOp> slice = new ArrayList<>(end - start);
+        for (int i = start; i < end; i++) {
+            slice.set(i, array.get(i));
+        }
+        return slice;
+    }
```

Comment:
I think it will only work till the start value is 0. If the start value is larger than 0, it will not behave as planned.